library/core/src/intrinsics.rs

```diff
@@ -2751,6 +2751,61 @@ pub const unsafe fn copy<T>(src: *const T, dst: *mut T, count: usize) {
     }
 }
 
+#[cfg(not(bootstrap))]
+extern "rust-intrinsic" {
+    /// This is an implementation detail of [`crate::mem::swap`] and should
+    /// not be used anywhere else.
+    ///
+    /// Swaps 2 values using minimal extra memory depending on target.
+    /// Created to remove target/backend specific optimizations from library code to
+    /// avoid confusing const evaluation and MIRI.
```

Comment:
The creation of this creates extra cost for CTFE and Miri, I don't understand the point of the comment here.